src/biogeophys/SoilTemperatureMod.F90

```diff
@@ -793,7 +802,14 @@ subroutine SoilThermProp (bounds,  num_nolakec, filter_nolakec, &
          do fc = 1,num_nolakec
             c = filter_nolakec(fc)
             if (snl(c)+1 < 1 .and. j >= snl(c)+1) then
-               cv(c,j) = cpliq*h2osoi_liq(c,j) + cpice*h2osoi_ice(c,j)
+!               cv(c,j) = cpliq*h2osoi_liq(c,j) + cpice*h2osoi_ice(c,j)
+!scs
+               if (frac_sno(c) > 0._r8) then
+                  cv(c,j) = max(1.0e-6_r8,(cpliq*h2osoi_liq(c,j) + cpice*h2osoi_ice(c,j))/frac_sno(c))
+               else
+                  cv(c,j) = 1.0e-6_r8
+               endif
+!scs
```

Comment:
Probably should check for frac_sno > small value rather than greater than zero.

Reply:
You are probably right but I think Sean should weigh in on this.

Reply:
Yes, that's fine.